pyntc/devices/ios_device.py

```diff
@@ -685,6 +685,109 @@ def file_copy_remote_exists(self, src, dest=None, file_system=None):
         log.debug("Host %s: File %s does not already exist on remote.", self.host, src)
         return False
 
+    def file_download(self, url, dest="", md5=None, file_system=None, read_timeout=2000):
+        """Download file from remote server.
+
+        Args:
+            url (str): URL of the file to download. (e.g., "scp://user:password@192.168.1.1/image.bin")
+            dest (str, optional): Destination name for file. Defaults to using the same name as the file on the remote server.
+            md5 (str, optional): Expected MD5 hash of the file to download. If provided, the file will be verified.
+            file_system (str, optional): File system to copy file to.
+            read_timeout (int, optional): Netmiko timeout when waiting for device prompt. Default 2000.
+
+        Raises:
+            SocketClosedError: Error raised if connection to device is closed.
+            FileTransferError: Error in transferring file.
+            FileTransferError: Error if unable to verify file was transferred successfully.
+        """
+        self.enable()
+        if file_system is None:
+            file_system = self._get_file_system()
+
+        filename = dest or os.path.basename(url)
+        try:
+            file_md5 = self.file_md5(filename, file_system)
+        except FileTransferError:
+            # File doesn't exist, download it
+            pass
+        else:
+            # File already exists
+            if not md5:
+                log.info("Host %s: File %s already exists on remote.", self.host, filename)
+                return True
+            if file_md5 != md5:
+                log.warning(
+                    "Host %s: File %s already exists on remote but MD5 hash mismatch. Downloading again.",
+                    self.host,
+                    filename,
+                )
+
+        copy_command = f"copy {url} {file_system}{dest}"
```

Comment:
Wondering if we should construct this with some validation.  I'm thinking file_system would be `bootflash` the dest you'd have to know to do `///image.bin` or `//something/image.bin`. Can we use `Path` or some other lib to try and construct this more intelligently?

Reply:
Even more so around the colon part.  Would expect `bootflash:///image.bin`